src/azure-cli-core/azure/cli/core/mock.py

```diff
@@ -17,16 +17,33 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
         from azure.cli.core.cloud import get_active_cloud
         from azure.cli.core.parser import AzCliCommandParser
         from azure.cli.core.util import random_string
-        from azure.cli.core._config import GLOBAL_CONFIG_DIR, ENV_VAR_PREFIX
+        from azure.cli.core._config import GLOBAL_CONFIG_DIR, ENV_VAR_PREFIX, ENV_VAR_TEST_LIVE
         from azure.cli.core._help import AzCliHelp
         from azure.cli.core._output import AzOutputProducer
 
         from knack.completion import ARGCOMPLETE_ENV_NAME
+        from knack.util import ensure_dir
+
+        # random_config_dir only takes effects in playback mode
+        if random_config_dir:
+            config_dir = os.path.join(GLOBAL_CONFIG_DIR, 'dummy_cli_config_dir', random_string())
+
+            # In recording mode, copy login credentials from global config dir to the dummy config dir
+            if os.getenv(ENV_VAR_TEST_LIVE, '').lower() == 'true':
+                if os.path.exists(GLOBAL_CONFIG_DIR):
+                    ensure_dir(config_dir)
+                    import shutil
+                    for file in ['azureProfile.json', 'msal_token_cache.bin', 'clouds.config']:
+                        try:
+                            shutil.copy(os.path.join(GLOBAL_CONFIG_DIR, file), config_dir)
+                        except FileNotFoundError:
+                            pass
```

Comment:
I am not very familiar with the knowledge in the field of auth, may I ask why do we pass for the `FileNotFoundError`?

Reply:
If the user is not logged in yet, these files do not exist in `~/.azure`. Skip copying them.